jose/jwt.go

```diff
@@ -14,30 +14,23 @@ func ParseJWT(token string) (jwt JWT, err error) {
 }
 
 func NewJWT(header JOSEHeader, claims Claims) (jwt JWT, err error) {
-	jwt = JWT{}
-
-	jwt.Header = header
-	jwt.Header[HeaderMediaType] = "JWT"
-
 	claimBytes, err := marshalClaims(claims)
 	if err != nil {
 		return
 	}
-	jwt.Payload = claimBytes
+	return newJWTFromPayload(header, claimBytes)
+}
 
-	eh, err := encodeHeader(header)
-	if err != nil {
-		return
-	}
-	jwt.RawHeader = eh
+func newJWTFromPayload(header JOSEHeader, payload []byte) (jwt JWT, err error) {
```

Comment:
The payload of a JWT must be JSON. This allows for the creation of malformed JWTs.

If this is meant to be a convenience method, the call order is backwards: newJWTFromPayload should be calling NewJWT (after constructing claims) instead of the other way around.